poc/two/load_student_data.py

```diff
@@ -0,0 +1,254 @@
+
+
+# Configure logging
+logging.basicConfig(
+    level=logging.INFO,
+    format='%(asctime)s - %(levelname)s - %(message)s'
+)
+logger = logging.getLogger(__name__)
+
+
+class EdFiApiClient:
+    """Client for interacting with Ed-Fi API"""
+    
+    def __init__(self, base_url: str, access_token: Optional[str] = None):
+        self.base_url = base_url.rstrip('/')
+        self.access_token = access_token
+        self.session = requests.Session()
+        
+        if access_token:
+            self.session.headers.update({
+                'Authorization': f'Bearer {access_token}',
+                'Content-Type': 'application/json'
+            })
+    
+    def authenticate(self, client_id: str, client_secret: str) -> bool:
+        """Authenticate with the Ed-Fi API and obtain access token"""
+        auth_url = f"{self.base_url}/oauth/token"
+        auth_data = {
+            'grant_type': 'client_credentials',
+            'client_id': client_id,
+            'client_secret': client_secret
+        }
+        
+        try:
+            response = requests.post(auth_url, data=auth_data)
+            response.raise_for_status()
+            
+            token_data = response.json()
+            self.access_token = token_data.get('access_token')
+            
+            self.session.headers.update({
+                'Authorization': f'Bearer {self.access_token}',
+                'Content-Type': 'application/json'
+            })
+            
+            logger.info("Successfully authenticated with Ed-Fi API")
+            return True
+            
+        except requests.RequestException as e:
+            logger.error(f"Authentication failed: {e}")
+            return False
+    
+    def create_student(self, student_data: Dict[str, Any]) -> bool:
+        """Create a student record in the Ed-Fi API"""
+        url = f"{self.base_url}/data/v3/ed-fi/students"
+        
+        try:
+            response = self.session.post(url, json=student_data)
+            
+            if response.status_code == 201:
+                logger.info(f"Successfully created student {student_data.get('studentUniqueId')}")
+                return True
+            elif response.status_code == 409:
+                logger.warning(f"Student {student_data.get('studentUniqueId')} already exists")
+                return True
+            else:
+                logger.error(f"Failed to create student {student_data.get('studentUniqueId')}: "
+                           f"{response.status_code} - {response.text}")
+                return False
+                
+        except requests.RequestException as e:
+            logger.error(f"Error creating student {student_data.get('studentUniqueId')}: {e}")
+            return False
+    
+    def create_enrollment(self, enrollment_data: Dict[str, Any]) -> bool:
+        """Create a student school enrollment record in the Ed-Fi API"""
+        url = f"{self.base_url}/data/v3/ed-fi/studentSchoolEnrollments"
```

Comment:
Hallucination! Not at all surprised that it did not figure out that enrollments are through `studentSchoolAssociations`. Again, the value of adding context.